src/Raven.Server/Documents/Indexes/VectorSearch/GenerateEmbeddings.cs

```diff
@@ -100,11 +143,55 @@ private static VectorValue Quantize(ByteStringContext allocator, in VectorEmbedd
         return embeddings;
     }
 
-    private static (IDisposable MemoryScope, Memory<byte> Memory, int UsedBytes) CreateEmbeddingViaSmartComponentsLocalEmbedding<TEmbedding>(ByteStringContext allocator, in string text, in int dimensions)
-        where TEmbedding : IEmbedding<TEmbedding>
+    private static (IDisposable MemoryScope, Memory<byte> Memory, int UsedBytes) CreateEmbeddingViaSmartComponentsLocalEmbedding(ByteStringContext allocator, in string text, in int dimensions)
+    {
+        List ??= new List<string>(1);
+        List.Add(text);
```

Comment:
Is that just an attempt to avoid an allocation? 

For that matter, we _know_ that it is more efficient to process that in bulk, no? Shouldn't we use that?

Reply:
yes, for that background task that we want to implement, no for local embedding generation - in indexes we process each document one by one